packages/eslint-plugin/src/rules/ban-ts-comment.ts

```diff
@@ -116,21 +126,66 @@ export default createRule<[Options], MessageIds>({
       }
     }
 
+    function execDirectiveRegEx(
+      regex: RegExp,
+      str: string,
+    ): MatchedTSDirective | null {
+      const match = regex.exec(str);
+      if (!match) {
+        return null;
+      }
+
+      const { directive, description } = nullThrows(
+        match.groups,
+        'RegExp should contain groups',
+      );
+      return {
+        directive: nullThrows(
+          directive,
+          'RegExp should contain "directive" group',
+        ),
+        description: nullThrows(
+          description,
+          'RegExp should contain "description" group',
+        ),
+      };
+    }
+
+    function findDirectiveInComment(
+      comment: TSESTree.Comment,
+    ): MatchedTSDirective | null {
+      if (comment.type === AST_TOKEN_TYPES.Line) {
+        const matchedPragma = execDirectiveRegEx(
+          singleLinePragmaRegEx,
+          `//${comment.value}`,
+        );
```

Comment:
Perhaps to be fully consistent with TypeScript's behavior, we should even check that this comment is in the begging of the file. Because `// @ts-nocheck` placed after some code is not a valid pragma. Sounds like a good followup, wdyt?

Reply:
Here it is - https://github.com/typescript-eslint/typescript-eslint/issues/8753